app/scripts/lib/stores/read-only-network-store.test.ts

```diff
@@ -0,0 +1,141 @@
+ *
+ * @param state
+ */
+function setMockFixtureServerReply(
+  state: Record<string, unknown> = DEFAULT_INITIAL_STATE,
+): void {
+  mockFixtureServerInterceptor().reply(200, state);
+}
+
+describe('ReadOnlyNetworkStore', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    nock.cleanAll();
+  });
+
+  describe('constructor', () => {
+    it('makes a request to the fixture server upon construction', async () => {
+      const scope = nock(FIXTURE_SERVER_ORIGIN)
+        .get(FIXTURE_SERVER_PATH)
+        .reply(200, 'Test');
+      expect(scope.pendingMocks().length).toBe(1);
+      setupReadOnlyNetworkStore();
+      await new Promise(process.nextTick);
+      expect(scope.pendingMocks().length).toBe(0);
+    });
+  });
+
+  describe('initialization (#init)', () => {
+    it('loads state from the network if fetch is successful and response is ok', async () => {
+      setMockFixtureServerReply(MOCK_STATE);
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toStrictEqual(MOCK_STATE);
+    });
+
+    it('does not throw, and logs a debug message, if fetch is not okay', async () => {
+      const logDebugSpy = jest
+        .spyOn(log, 'debug')
+        .mockImplementation(() => undefined);
+
+      mockFixtureServerInterceptor().reply(400);
+
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toBe(null);
+      expect(logDebugSpy).toHaveBeenCalledWith(
+        'Received response with a status of 400 Bad Request',
+      );
+    });
+
+    it('does not throw, and logs a debug message, if fetch throws an error', async () => {
+      mockFixtureServerInterceptor().replyWithError('error!');
+      const logDebugSpy = jest
+        .spyOn(log, 'debug')
+        .mockImplementation(() => undefined);
+
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toBe(null);
+      expect(logDebugSpy).toHaveBeenCalledWith(
+        "Error loading network state: 'request to http://localhost:12345/state.json failed, reason: error!'",
+      );
+    });
+  });
+
+  describe('get', () => {
+    it('returns null if #state is null', async () => {
+      mockFixtureServerInterceptor().reply(200);
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toBe(null);
+    });
+
+    it('returns stored state if #state has data', async () => {
+      setMockFixtureServerReply(MOCK_STATE);
+
+      const store = setupReadOnlyNetworkStore();
+      const result = await store.get();
+      expect(result).toStrictEqual(MOCK_STATE);
+    });
+  });
+
+  describe('set', () => {
+    it('throws if not passed a state parameter', async () => {
+      const store = setupReadOnlyNetworkStore();
+      await expect(
+        store.set(undefined as unknown as MetaMaskStorageStructure),
```

Comment:
Nit: Recommended practice for handling "intentional type errors" in tests is to use `@ts-expect-error` and explain why it's invalid.

Docs here: https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#use-ts-expect-error-to-force-runtime-execution-of-a-branch-for-validation-or-testing

```suggestion
        // @ts-expect-error Intentionally passing incorrect type
        store.set(undefined),
```

Generally speaking, we should be using the `as` operator (unsafe upcast) in places where we want to transform the type in a way TypeScript can't guarantee. That kinda describes the current scenario: we are passing in `undefined` and we want it to be `MetaMaskStorageStructure`. But it's not really true that we want to "transform" it; both types are already correct for their given contexts.

Really we expect this to be an error, we just want to override it. "// @ts-expect-error" is perfectly suited for this, and has the added bonus feature of letting us know when the error override is no longer needed.

Reply:
suggestion applied in 1a2ce8fa0b